From: Michael Froman <mfroman@mozilla.com>
Date: Mon, 19 Jan 2026 14:40:52 -0600
Subject: (tmp-cherry-pick) Revert "sdk: add RTC_OBJC_TYPE to ios Metal
 renderer classes" (81920f654b)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This reverts commit 89e8619bfdacfd1364bf3fc6bc5388c92629ab71.

Reason for revert: Breaks ios dependencies, including AppRTCMobile_ios_test

Original change's description:
> sdk: add RTC_OBJC_TYPE to ios Metal renderer classes
>
> Bug: webrtc:460166677
> Change-Id: Id8883d40957e26046897632c4f6ed2d9c36bcf4e
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/427320
> Reviewed-by: Kári Helgason <kthelgason@webrtc.org>
> Commit-Queue: Kári Helgason <kthelgason@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46309}

Bug: webrtc:460166677
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Change-Id: I287daaf0c65942bc43579f32d38aba4d9ef4ac00
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/428622
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: Evan Shrubsole <eshr@webrtc.org>
Commit-Queue: Evan Shrubsole <eshr@webrtc.org>
Reviewed-by: Jeremy Leconte <jleconte@google.com>
Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#46310}
---
 sdk/objc/components/renderer/metal/RTCMTLI420Renderer.h | 2 +-
 sdk/objc/components/renderer/metal/RTCMTLNV12Renderer.h | 2 +-
 sdk/objc/components/renderer/metal/RTCMTLRGBRenderer.h  | 2 +-
 sdk/objc/components/renderer/metal/RTCMTLRenderer.h     | 3 +--
 4 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/sdk/objc/components/renderer/metal/RTCMTLI420Renderer.h b/sdk/objc/components/renderer/metal/RTCMTLI420Renderer.h
index 5a1e6e3d51..e5987fe22a 100644
--- a/sdk/objc/components/renderer/metal/RTCMTLI420Renderer.h
+++ b/sdk/objc/components/renderer/metal/RTCMTLI420Renderer.h
@@ -13,5 +13,5 @@
 #import "RTCMTLRenderer.h"
 
 NS_AVAILABLE(10_11, 9_0)
-@interface RTC_OBJC_TYPE (RTCMTLI420Renderer) : RTCMTLRenderer
+@interface RTCMTLI420Renderer : RTCMTLRenderer
 @end
diff --git a/sdk/objc/components/renderer/metal/RTCMTLNV12Renderer.h b/sdk/objc/components/renderer/metal/RTCMTLNV12Renderer.h
index 6ac6653493..866b7ea17e 100644
--- a/sdk/objc/components/renderer/metal/RTCMTLNV12Renderer.h
+++ b/sdk/objc/components/renderer/metal/RTCMTLNV12Renderer.h
@@ -13,6 +13,6 @@
 #import "RTCMTLRenderer.h"
 
 NS_AVAILABLE(10_11, 9_0)
-@interface RTC_OBJC_TYPE (RTCMTLNV12Renderer) : RTCMTLRenderer
+@interface RTCMTLNV12Renderer : RTCMTLRenderer
 
 @end
diff --git a/sdk/objc/components/renderer/metal/RTCMTLRGBRenderer.h b/sdk/objc/components/renderer/metal/RTCMTLRGBRenderer.h
index 8f4fb1b78e..9db422cd22 100644
--- a/sdk/objc/components/renderer/metal/RTCMTLRGBRenderer.h
+++ b/sdk/objc/components/renderer/metal/RTCMTLRGBRenderer.h
@@ -17,6 +17,6 @@
  * kCVPixelFormatType_32ARGB.
  */
 NS_AVAILABLE(10_11, 9_0)
-@interface RTC_OBJC_TYPE (RTCMTLRGBRenderer) : RTCMTLRenderer
+@interface RTCMTLRGBRenderer : RTCMTLRenderer
 
 @end
diff --git a/sdk/objc/components/renderer/metal/RTCMTLRenderer.h b/sdk/objc/components/renderer/metal/RTCMTLRenderer.h
index ea2bb73859..84aa3fe7fc 100644
--- a/sdk/objc/components/renderer/metal/RTCMTLRenderer.h
+++ b/sdk/objc/components/renderer/metal/RTCMTLRenderer.h
@@ -21,8 +21,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * Protocol defining ability to render RTCVideoFrame in Metal enabled views.
  */
-@protocol RTC_OBJC_TYPE
-(RTCMTLRenderer)<NSObject>
+@protocol RTCMTLRenderer <NSObject>
 
 /**
  * Method to be implemented to perform actual rendering of the provided frame.
